### PR TITLE
ResponseMatcher created for #153

### PR DESCRIPTION
**Pull request by olenagerasimova**
Created `ResponseMatcher` for #153 with a bunch of ctors and used it a couple of test classes.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#155](https://github.com/artipie/http/issues/155) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/2ef765fb-b33d-4992-90d5-127f119dd9cf); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/http/hm/ResponseMatcher.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import com.artipie.http.rs.RsStatus;
+import org.cactoos.list.ListOf;
+import org.hamcrest.Matcher;
+import org.hamcrest.core.AllOf;
+
+/**
+ * Response matcher.
+ * @since 0.10
+ */
+public final class ResponseMatcher extends AllOf<Response> {
```
Comment (carlosmiranda) [inline src/main/java/com/artipie/http/hm/ResponseMatcher.java:37]: @olenagerasimova we should have a test for the matcher as well. That way we shouldn't need to bump down the coverage.

**Comment by olenagerasimova** [comment]
@carlosmiranda thanks, done, take a look please

**Comment by olenagerasimova** [comment]
@g4s8 could you please merge this?

File: src/main/java/com/artipie/http/hm/ResponseMatcher.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            )
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param status Expected status
+     * @param body Expected body
+     */
+    public ResponseMatcher(final RsStatus status, final String body) {
```
Comment (g4s8) [inline src/main/java/com/artipie/http/hm/ResponseMatcher.java:74]: @olenagerasimova I'd ask to specify the `Charset`, without it this matcher will be platform and environment dependent  and may produce different bytes from same string depends on system charset encoding

File: src/main/java/com/artipie/http/hm/ResponseMatcher.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                new RsHasBody(body.getBytes())
+            )
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param body Expected body
+     */
+    public ResponseMatcher(final String body) {
```
Comment (g4s8) [inline src/main/java/com/artipie/http/hm/ResponseMatcher.java:87]: @olenagerasimova same here: `body.getBytes()` may return different array on different environment

**Comment by olenagerasimova** [comment]
@g4s8 thanks, added charsets

**Comment by g4s8** [comment]
@olenagerasimova git message linter failed: please format your commit messages in format `#<ticket-number> - <message>` and force push to your branch

**Comment by olenagerasimova** [comment]
@g4s8 sorry, but I've spoiled this branch totally by pushing to many commits. I do not think we should ever merge this, so I've opened #160. Can you please make sure that @carlosmiranda will get paid for the review here?

**Comment by 0crat** [comment]
Job `gh:artipie/http#155` is not assigned, can't get performer

**Pull request closed without merging**